### ISSUE TITLE
Property collector error: context deadline exceeded on container start

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#2372]
Saw this issue on a recent functional run on a Nimbus VC cluster with 3 ESX hosts. Probably related to #1822 , but I am going to leave that closed for the time being.

From portlayer.log:
```
time="2016-09-19T15:14:58Z" level=debug msg="Found 1 subscribers to vsphere.VmEvent: fervent_bhaskar-1379b2830c2c417e926a0fa924bb6c4f60d432443a4dae6c5085eae25a924648 on  10.160.120.204 in ha-datacenter is powered off" 
time="2016-09-19T15:14:58Z" level=debug msg="Event manager calling back to exec" 
time="2016-09-19T15:14:58Z" level=debug msg="Container(1379b2830c2c417e926a0fa924bb6c4f60d432443a4dae6c5085eae25a924648) state set to Stopped via event activity" 
time="2016-09-19T15:14:58Z" level=debug msg="Container(1379b2830c2c417e926a0fa924bb6c4f60d432443a4dae6c5085eae25a924648) closing 0 log followers" 
time="2016-09-19T15:14:58Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.(*Container).Update] Container.Update" 
time="2016-09-19T15:17:44Z" level=error msg="Property collector error: context deadline exceeded" 
```

From docker-personality.log:
```
time="2016-09-19T15:14:42Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerStart] 1379b2830c2c417e926a0fa924bb6c4f60d432443a4dae6c5085eae25a924648" 
time="2016-09-19T15:17:44Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerStart] [3m2.034466734s] 1379b2830c2c417e926a0fa924bb6c4f60d432443a4dae6c5085eae25a924648" 
time="2016-09-19T15:17:44Z" level=error msg="Handler for POST /v1.23/containers/1379b2830c2c417e926a0fa924bb6c4f60d432443a4dae6c5085eae25a924648/start returned error: Server error from portlayer: unable to wait for process launch status: context deadline exceeded" 
```

[Test-Cases.Group1-Docker-Commands.1-11-Docker-RM-VCH-0-7900-container-logs.zip](https://github.com/vmware/vic/files/480885/Test-Cases.Group1-Docker-Commands.1-11-Docker-RM-VCH-0-7900-container-logs.zip)

**Comment by mhagen-vmware** [comment]
Note: when attempting to start a container, it might fail with context deadline exceeded especially on a loaded system.  You can wait until the system is less loaded and retry.

**Comment by jzt** [comment]
I believe @dougm's PR (#2405) addresses this issue.

While looking at the logs attached to this ticket, @hmahmood and I found that the guestinfo key for `started` was never set (see @dougm's problem description in #2327). Since it was a short-lived container (a busybox `/bin/ls`), the vm was already powered off by the time `WaitForKeyInExtraConfig` was called. This caused the 3-minute context timeout to be reached, producing the CDE we saw in this integration test. @dougm's PR fixes this by not allowing the property collector to sit and wait if the VM is powered off.

**Comment by dougm** [comment]
Not quite, my PR just quits `WaitForKeyInExtraConfig` earlier than the context timeout if the vm powers off, still results in an error.  I don't see any container .debug logs in the attached .zip either.

**Comment by jzt** [comment]
@dougm sorry I should have said that your PR will prevent the CDEs from occurring. The root cause is captured in #2327 under the `Tether needs to set the SessionConfig.Started field`.

I guess this issue should be considered a related issue to #2327.

**Comment by mhagen-vmware** [comment]
https://ci.vmware.run/vmware/vic/4973:

[integration_logs_4973_f84b25ced9b22d948ea63242a02880d5c0c489fd_1474489843.zip](https://github.com/vmware/vic/files/486014/integration_logs_4973_f84b25ced9b22d948ea63242a02880d5c0c489fd_1474489843.zip)

**Comment by dougm** [comment]
Fixed by #2327

At the very least, 'docker start' should no longer CDE.  If tether fails to initialize, the error message will indicate powerState=poweredOff and any tether init failures would be in the container .debug log.

**Comment by stuclem** [comment]
This is fixed. Removing kind/note.
